apps/api/src/common/auth.middleware.ts

```diff
@@ -25,7 +25,7 @@ export class AuthenticationMiddleware implements NestMiddleware {
       throw new HttpException('Token not found', HttpStatus.BAD_GATEWAY);
     }
     try {
-      const user: any = await this.authService.getUserFromToken(token);
+      const user = await this.authService.getUserFromToken(token);
```

Comment:
Just a question though it's not about this PR.
It seems that both of auth guard and middleware validate the token and set the user. To separate concerns, can we make middleware validate token and guard validate role?

Reply:
I feel we can move both into auth guard. As In reporting we might have roles that are separate from the existing role with very limited functionality.
AuthGuard will be a simple solution here, As if we do want to check partial URLs from the middleware then it will add an overhead for a developer to keep track of it, It needs to be updated for each URL which should check user token.